src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -36,12 +37,55 @@ protected override Task<ImmutableArray<ISymbol>> GetSymbolsAsync(SyntaxContext c
                 context.SemanticModel, position, context.Workspace, options, cancellationToken);
         }
 
+        protected async override Task<bool> ShouldProvidePreselectedItemsAsync(CompletionContext completionContext, SyntaxContext syntaxContext, Document document, int position, Lazy<ImmutableArray<ITypeSymbol>> inferredTypes, OptionSet options)
+        {
+            var sourceText = await document.GetTextAsync(CancellationToken.None).ConfigureAwait(false);
+            if (ShouldTriggerInArgumentLists(sourceText, options))
+            {
+                // Avoid preselection & hard selection when triggered via insertion in an argument list.
+                if (completionContext.Trigger.Kind == CompletionTriggerKind.Insertion &&
+                    position > 0 &&
+                    await IsTriggerInArgumentListAsync(document, position - 1, CancellationToken.None).ConfigureAwait(false) == true)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
         protected override bool IsInstrinsic(ISymbol s)
             => s is ITypeSymbol ts && ts.IsIntrinsicType();
 
         internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
         {
-            return CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
+            return ShouldTriggerInArgumentLists(text, options)
+                ? CompletionUtilities.IsTriggerCharacterOrArgumentListCharacter(text, characterPosition, options)
+                : CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
```

Comment:
consdier actually writing this as:

```
if (IsTrigger()) return true;
return ShouldTriggerInArgumentLists() && IstrigerInArgumentLists
```

Reply:
I'm seeing them as basically equivalently easy to read. Going to leave alone unless you have stronger objections.